7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R

```diff
@@ -1,6 +1,113 @@
+
+  dat_raw <- readr::read_csv(infile)
+
+  # split date and time
+  dat_datetime_split <- stringr::str_split_fixed(dat_raw$Timestamp, " ", 2) %>%
+    as_tibble %>%
+    rename('DateTime' = V1, 'time' = V2)
+
+  data_clean <- dat_raw %>%
+    cbind(., dat_datetime_split) %>%
+
+    dplyr::mutate(DateTime = as.Date(DateTime),
+           depth = convert_ft_to_m(Depth),
+           temp = fahrenheit_to_celsius(`Temperature (F)`),
+           Missouri_ID = 'Missouri_100', # used for crosswalk
+           site = Site) %>%
+    dplyr::select(DateTime, time, depth, temp, Missouri_ID, site)
+
+  saveRDS(object = data_clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+
+}
+
+parse_Bull_Shoals_and_LOZ_profile_data_LMVP <- function(inind, outind) {
+
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  dat_raw <- readxl::read_xlsx(infile)
+
+  data_clean <- dat_raw %>%
+    dplyr::mutate(DateTime = as.Date(Date),
+                  depth = `Depth m`,
+                  temp = C,
+                  Missouri_ID = Lake,
+                  site = `Name and Site`,
+                  Missouri_ID = case_when(
+                    # values from Univ MO xwalk (https://drive.google.com/file/d/11w6-LXCDSDCjipFYPxUgJyf7YB9BtXYR/view?usp=sharing)
+                    Lake == 'Bull Shoals' ~ 'Missouri_100',
+                    Lake == 'Lake of the Ozarks' ~ 'Missouri_149'
+                  )
+                ) %>%
+    dplyr::select(DateTime, depth, temp, Missouri_ID, site)
+
+  saveRDS(object = data_clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+
+}
+
+
+# Parse the 14 files in Temp_DO_BSL_MM_DD_YYY
+parse_Temp_DO_BSL_MM_DD_YYYY <- function(inind, outind) {
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  # Unzip the file with all of Bull Shoals, then cleanup unzipped files
+  # which aren't needed externally before leaving the function.
+  unzip_dir <- tempdir()
+  files_from_zip <- unzip(infile, exdir = unzip_dir)
+  on.exit(unlink(unzip_dir, recursive = TRUE))
+
+  data_clean <- files_from_zip %>%
+    purrr::map_df(~ read_bullshoals_data(.,
+                                         new_col_names = c('DateTime', 'DO',
+                                                           'temp', 'site',
+                                                           'depth', 'unit_id')))
+
+  saveRDS(object = data_clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+}
+
+read_bullshoals_data <- function(file_path, new_col_names, keep_cols) {
+
+  # identify sheets with data
+  list_sheets <- readxl::excel_sheets(file_path) %>%
+    .[grepl('^Pt', .)]
+
+  # read in data and clean
+  clean <- list_sheets %>%
+    purrr::map_df(~ readxl::read_xlsx(path = file_path, sheet = .x)) %>%
+    purrr::discard(~all(is.na(.x))) %>%  # remove columns with all NA
```

Comment:
WOW!! I didn't know about `purrr::discard()` or `purrr::keep()`. Can't wait to use those in the future!

Reply:
Glad to hear! I'm working on replacing my dependence on the `apply` family with `purrr` so I'm definitely digging into the doc.